testing/ui-tests/src/lib.rs

```diff
@@ -15,14 +15,15 @@ mod dispatch_errors;
 mod storage;
 mod utils;
 
-use crate::utils::MetadataTestRunner;
+use crate::utils::{MetadataTestRunner, PalletMetadataTestRunner};
 
 // Each of these tests leads to some rust code being compiled and
 // executed to test that compilation is successful (or errors in the
 // way that we'd expect).
 #[test]
 fn ui_tests() {
     let mut m = MetadataTestRunner::default();
+    let mut p = PalletMetadataTestRunner::new();
```

Comment:
Nit: maybe move this to be right above that while loop where it's used?

Reply:
Yep, that was my intuition too, although it needs to be kept around until after the test suite is dropped. Otherwise, we'll delete the test files before the test suite gets to running them. Apparently, the UI tests are running once the test suite object is dropped

Reply:
Ah interesting; yeah that rings a bell actually! very strange :)